core/src/main/java/com/nisovin/magicspells/spells/PlayerMenuSpell.java

```diff
@@ -176,16 +209,18 @@ private void processClickSpell(Subspell subspell, Player caster, Player target,
 		subspell.cast(caster, power);
 	}
 
-	private void open(Player opener, SpellData data) {
+	private void open(Player opener, MenuData data, int page) {
+		SpellData spellData = data.spellData();
+
 		List<Player> players = new ArrayList<>(Bukkit.getOnlinePlayers());
 		if (!addOpener) players.remove(opener);
 		if (playerModifiers != null) players.removeIf(player -> !playerModifiers.check(player));
 		if (radius > 0) players.removeIf(player -> opener.getLocation().distance(player.getLocation()) > radius);
 
-		int size = (int) Math.ceil((players.size()+1) / 9.0) * 9;
+		int size = (int) Math.ceil(Math.min((players.size()+1), 54) / 9.0) * 9;
```

Comment:
Unrelated to your changes, but the addition of one in `(players.size()+1)` should probably be removed.